src/toil_scripts/lib/test/test_programs.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+
+def test_which():
+    from toil_scripts.lib.programs import which
+    assert which('python')
+
+
+def test_docker_call(tmpdir):
+    from toil_scripts.lib.programs import docker_call
+    work_dir = str(tmpdir)
+    parameter = ['--help']
+    tool = 'quay.io/ucsc_cgl/samtools'
+    docker_call(work_dir=work_dir, parameters=parameter, tool=tool)
+    # Test outfile
+    fpath = os.path.join(work_dir, 'test')
+    with open(fpath, 'w') as f:
+        docker_call(tool='ubuntu', env=dict(foo='bar'), parameters=['printenv', 'foo'], outfile=f)
+    assert open(fpath).read().strip() == 'bar'
```

Comment:
Removing strip() and comparing against 'bar\n' would be a stronger assertion. It's also shorter.